hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -1899,6 +1899,10 @@ synchronized void transitionToStandby() throws IOException {
   synchronized void transitionToObserver() throws IOException {
     String operationName = "transitionToObserver";
     namesystem.checkSuperuserPrivilege(operationName);
+    if (namesystem.isInSafeMode()) {
+      throw namesystem.newSafemodeException("Cannot transition to " +
+          OBSERVER_STATE);
```

Comment:
Consolidate this logic with the exception from `transitionToActive`:
```java
    if (notBecomeActiveInSafemode && isInSafeMode()) {
      throw new ServiceFailedException(getRole() + " still not leave safemode");
    }
```
I don't think we need to make `newSafemodeException` public, seems fine to just throw a new exception here?